presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -700,7 +700,16 @@ protected Scope visitUnnest(Unnest node, Optional<Scope> scope)
                 ExpressionAnalysis expressionAnalysis = analyzeExpression(expression, createScope(scope));
                 Type expressionType = expressionAnalysis.getType(expression);
                 if (expressionType instanceof ArrayType) {
-                    outputFields.add(Field.newUnqualified(Optional.empty(), ((ArrayType) expressionType).getElementType()));
+                    Type elementType = ((ArrayType) expressionType).getElementType();
+                    if (elementType instanceof RowType) {
+                        outputFields.addAll(
+                                elementType.getTypeParameters().stream()
+                                        .map(type -> Field.newUnqualified(Optional.empty(), type))
+                                        .collect(toImmutableList()));
```

Comment:
I personally prefer using a for-loop here. If you prefer `stream`, you could do `                               .forEach(outputFields::add)`. This way, you wouldn't need `outputFields.addAll` any more. You can change your code either way.